### PR TITLE
fix(dropdown-menu): Call onMenuClose on document clicks

### PR DESCRIPTION
**Pull request by sahiga**
This PR modifies `handleDocumentClick()` in `DropdownMenu` to call the passed-in `onMenuClose()` function (if provided) in addition to `closeMenu()`.

### CODE REVIEW

File: src/components/dropdown-menu/DropdownMenu.js [hunk 1]
```diff
@@ -168,7 +168,13 @@ class DropdownMenu extends React.Component<Props, State> {
             !menuEl.contains(event.target) &&
             !menuButtonEl.contains(event.target)
         ) {
+            const { onMenuClose } = this.props;
+
             this.closeMenu();
+
+            if (onMenuClose) {
```
Comment (ConradJChan) [inline src/components/dropdown-menu/DropdownMenu.js]: Could we just default onMenuClose to noop?

Comment (sahiga) [reply src/components/dropdown-menu/DropdownMenu.js]: @ConradJChan I just updated this file with your suggestion. Could you review again? Thanks!

File: src/components/dropdown-menu/DropdownMenu.js [hunk 2]
```diff
@@ -168,7 +169,9 @@ class DropdownMenu extends React.Component<Props, State> {
             !menuEl.contains(event.target) &&
             !menuButtonEl.contains(event.target)
         ) {
+            const { onMenuClose = noop } = this.props;
```
Comment (ConradJChan) [inline src/components/dropdown-menu/DropdownMenu.js]: Perhaps we could move this to `defaultProps` and then simplify `handleMenuClose` too?

Comment (sahiga) [reply src/components/dropdown-menu/DropdownMenu.js]: Great idea. Will fix

Comment (ConradJChan) [reply src/components/dropdown-menu/DropdownMenu.js]: Seems like defaulting to noop in defaultprops should be good enough? Is the default when destructuring needed?

Comment (sahiga) [reply src/components/dropdown-menu/DropdownMenu.js]: I ran into a bunch of Flow issues when I tried to add `onMenuClose` as a default prop while also keeping it optional, so I resorted to destructuring instead.

**Pull request merged**